src/components/LeftSideBar.tsx

```diff
@@ -1,42 +1,103 @@
 import * as React from 'react';
+import * as $ from 'jquery';
+
 import { ToastContainer } from 'react-toastify';
 import VarList from './VarList';
 import GMList from './GMList';
 import TemplateList from './TemplateList';
 
-/**
- * A left sidebar container that holds the variable, graphics methods and templates lists.
- * 
- * props: 
- *  varClickHandler: a function that handles when an item in the variables list is clicked.
- *  variables: A dictionary containing variable names and values
- *  graphicMethods: A dictionary conatining graphic method names and values
- *  templates: A dictionary conatining template names and values
- */
 
-export class LeftSideBar extends React.Component <any> {
-    
+declare var vcs: any;
+
+type LeftSideBarProps = {};
+type LeftSideBarState = {
+    file_path: string,          // path to the file in question
+    variables: any,             // object of varnames: varinfo
+    graphicMethods: any,        // list of available graphicsmethods
+    templates: any,             // list of availabe templates
+    showEditVariable: boolean,  // should the edit variable modal show up
+    active_variable: string     // name of the selected variable
+};
+
+export class LeftSideBar extends React.Component<LeftSideBarProps, LeftSideBarState> {
+
+    canvas: any;
+    console: any;
+    varListEl: any;
+    constructor(props: LeftSideBarProps) {
+        super(props);
+        this.state = {
+            file_path: '',
+            variables: [],
+            graphicMethods: [],
+            templates: [],
+            showEditVariable: false,
+            active_variable: '',
+        }
+        this.varListEl = (React as any).createRef();
+
+        this.handleVCSLoad = this.handleVCSLoad.bind(this);
+        this.handleVarClick = this.handleVarClick.bind(this);
+        this.handleLoadVariable = this.handleLoadVariable.bind(this);
+        this.handleLoadFile = this.handleLoadFile.bind(this);
+    }
+    // initialize the vcs.js object and pass it the children that need it
+    handleVCSLoad() {
+        console.log('vcs.js load complete');
+        let vcs_target = $('#vcs-target')[0];
+        this.canvas = vcs.init(vcs_target);
+        this.varListEl.current.setupVcs(vcs);
+    }
+    // setup the global vcs.js object
+    componentDidMount() {
+        console.log('starting vcs.js load');
+        let script = document.createElement('script');
+        script.src = `http://localhost:5000/vcs.js`;
+        script.async = true;
+        script.addEventListener('load', this.handleVCSLoad);
+        document.body.appendChild(script);
+    }
+    // handle click on a variable by opening dimension select modal
+    handleVarClick(e: any) {
+        debugger;
+    }
+    handleSetupConsole(console: any) {
+        this.console = console;
+        let script = [
+            'import cdms2',
+            'import vcs',
+            'x = vcs.init()'];
```

Comment:
@downiec that is probably where you want to add your function to query variables and and graphic methods.

Reply:
This is the setup script that gets run once on startup.